Code/Framework/AzCore/AzCore/Math/LineSegment.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/RTTI/TypeInfoSimple.h>
+
+namespace O3DE
+{
+    class LineSegment
+    {
+    public:
+        AZ_TYPE_INFO(LineSegment, "{7557da1e-cc20-11ec-9d64-0242ac120002}");
+
+        explicit LineSegment(const AZ::Vector3& start, const AZ::Vector3& end);
+
+        const AZ::Vector3& GetStart() const;
+        const AZ::Vector3& GetEnd() const;
+        
+        LineSegment& operator=(const LineSegment& rhs);
+        bool operator==(const LineSegment& rhs) const;
+        bool operator!=(const LineSegment& rhs) const;
+
+        //TODO: implement
+        //!
+        float GetLength();
+    
+        //!
+        AZ::Vector3 GetDirection();
+        
+        //!
+        AZ::Vector3 GetPoint(float t);
+        
+        //!
+        bool IsPointOnSegment(const AZ::Vector3& point);
```

Comment:
that's probably a good strategy since you end up implementing it in both cases. LineSegment will implement line segment to sphere and sphere would implement the opposite case. not sure if it make sense to provide those utility methods?